src/riak_kv_ts_util.erl

```diff
@@ -669,17 +669,35 @@ check_table_feature_supported(DDLRecCap, DecodedReq) ->
                     {error, create_table_not_supported_message(Table)}
             end;
         {ok, _, {_, Table}} ->
-            case is_table_supported(DDLRecCap, Table) of
+            case is_table_active_and_supported(DDLRecCap, Table) of
                 true ->
                     DecodedReq;
                 false ->
-                    {error, table_not_supported_message(Table)}
+                    {error, table_not_supported_message(Table)};
+                Error ->
+                    Error
             end;
         _ ->
             DecodedReq
     end.
 
 %%
+-spec is_table_active_and_supported(DDLRecCap::atom(),
+                                    Table::binary()) ->
+                                        boolean() | {error, string()}.
+is_table_active_and_supported(DDLRecCap, Table) ->
+    Mod = riak_ql_ddl:make_module_name(Table),
+    case code:ensure_loaded(Mod) of
```

Comment:
Yes, the bucket doesn't exist in testing. @andytill took a different approach in https://github.com/basho/riak_kv/pull/1530, passing a function to allow for mocking. I didn't find the approach complicated, but the mock provided IMHO an overly simplified bucket properties response. Down that rabbit hole, I decided that simplifying the mocking to ensure the full range of return types up a level achieved the same goal in a more straightforward manner. The point of the mock is to isolate the functionality under unit test to just the unit under development. The fact that a TS table creation results in these invariants is tested elsewhere.